packages/api/src/interface.ts

```diff
@@ -17,3 +17,17 @@ export type Api = {
   node: NodeApi;
   validator: ValidatorApi;
 };
+
+// Declare namespaces for CLI options
+export type ApiNamespace = keyof Api;
+const allNamespacesObj: {[K in keyof Api]: true} = {
```

Comment:
Annoying that we had to do this, but I understand why. `allNamespacesObj`, and its `K in keyof...` type is what enforces the connection between the namespace list as a type (`Api`), and the namespace list as a javascript object/array.